Improved custom indexer code and description (#3465)
magento/devdocs: guides/v2.1/extension-dev-guide/indexing-custom.md

To push best-selling products to the top of a category listing, process statistics about sales to change the product position dynamically.

> Assuming your module is named `<VendorName>_Merchandizing`, you must write the appropriate code in the indexer class `Merchandizing/Model/Indexer/Test.php` 

Reviewer: ```suggestion
Assuming your module is named `<VendorName>_Merchandizing`, you must write the appropriate code in the indexer class `Merchandizing/Model/Indexer/Test.php` 
```
Not should use `Merchandizing/Model/Indexer/Popular.php`
based on the name of the class below:
`class Popular implements \Magento\Framework\Indexer\ActionInterface, \Magento\Framework\Mview\ActionInterface`

Author: Yeah. I got it.

    }

    /*
     * Will take all of the data and reindex
     * Will run when reindex via command line
     */
    public function executeFull(){
        //Should take into account all placed orders in the system
    }

    /*
     * Works with a set of entity changed (may be massaction)
     */
    public function executeList(array $ids){
        //Works with a set of placed orders (mass actions and so on)
    }

    /*
     * Works in runtime for a single entity using plugins
     */
    public function executeRow($id){
        //Works in runtime for a single order using plugins
    }
}
```

Next, declare the indexer in `Merchandizing/etc/indexer.xml`:

```xml
<?xml version="1.0"?>
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Indexer/etc/indexer.xsd">
    <indexer id="merchandizing_popular" view_id="merchandizing_popular_order" class="Vendor\Merchandizing\Model\Indexer\Popular">
        <title translate="true">Popular Products</title>
        <description translate="true">Sort products in a category by popularity</description>
    </indexer>
</config>
```

> In this file, Need to declare a new indexer process with the attribute:

Reviewer 2: Recommended edit: 

```suggestion
In this file, declare a new indexer process with the attribute:
```

- `id` - To identify this indexer. Check status, mode or reindex this indexer by command line.
- `view_id` - Id of view element which will be defined in the `mview` configuration file.
- `class` -  The name to the class which we process indexer method.

Finally, declare the indexer view (`merchandizing_popular_order`) that tracks sales (`Merchandizing/etc/mview.xml`):

```xml
<?xml version="1.0" encoding="UTF-8"?>
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Mview/etc/mview.xsd">
    <view id="merchandizing_popular_order" class="Vendor\Merchandizing\Model\Indexer\Popular" group="indexer">
        <subscriptions>
            <table name="sales_order" entity_column="entity_id" />
        </subscriptions>
    </view>
</config>
```

These settings start `<VendorName>\Merchandizing\Model\Indexer\Popular::execute` method every time an order is changed.

> After this, please refresh the cache and go to **System > Index Management** from backend to see custom indexer the result.

Reviewer 2: Slight edit, adding some doc links:

```suggestion
After this, please refresh the cache and go to **System > [Index Management](https://docs.magento.com/m2/ce/user_guide/system/index-management.html)** through the Admin to verify the custom indexer result.
```

> ![Custom indexer modes]({{ site.baseurl }}/common/images/custom-indexer-magento2.jpg){:width="600px"}

Reviewer 2: Recommend adding a blank line, it helps on publish for images and notes. :) 

```suggestion

![Custom indexer modes]({{ site.baseurl }}/common/images/custom-indexer-magento2.jpg){:width="600px"}
```
